lib/puppet/file_system/windows.rb

```diff
@@ -134,10 +137,30 @@ def replace_file(path, mode = nil)
              dacl
            when 0640, 0600
              secure_dacl(current_sid)
+           when 0660
+             dacl = secure_dacl(current_sid, FILE_READ_WRITE, FILE_READ_WRITE)
+             dacl
+           when 0664
+             dacl = secure_dacl(current_sid, FILE_READ_WRITE, FILE_READ_WRITE)
+             dacl.allow(Puppet::Util::Windows::SID::BuiltinUsers, FILE_READ)
+             dacl.allow(Puppet::Util::Windows::SID::Everyone, FILE_READ)
+             dacl
```

Comment:
I would drop `Everyone` and only set `BuiltUsers` to read or readwrite, as that's what Windows does when creating a file generally. Same comment below.